### PR TITLE
First step to make the code more pythonic

### PR DESCRIPTION
**Pull request by dduong42**
In this PR:
- Remove some getters/setters.
- Remove change some `if len(l)` into `if l`
- Change the unused variables into `_` when unpacking.
- Remove some unused code.
- Add the `ignore_curse_errors` context manager (avoid the `try`/`pass`)
- Use `values` with a dictionary when the key is not needed.
- Remove the `controller` attribute from the `SimpleLine` and the `LineMatch`. (Pass it to `output` instead).
- ...

To sum up, this a first step to make the code a little bit more pythonic. There is some work left, but I didn't want to make a PR with too many commits. The `format.py` file should be a little less "java".

### CODE REVIEW

**Comment by pcottle** [comment]
Wow this is huge @dduong42 ! I don't write much python at all so a lot of the code is nowhere near pythonic (as you can probably tell)

Before I take a closer look at this, the function decorators are supported the same in python 3.0 right? We are close to getting python 3.0 support so don't want to regressin that sense

**Comment by dduong42** [comment]
The changes won't affect the python 3 compatibility :).

**Comment by zkanda** [comment]
I suggest breaking this into multiple pr's, being more pythonic is very subjective.

But I agree, using `property` decorator is very pythonic. :)

**Comment by mahamudul007** [comment]
Duoss4462@gmail.com

**Comment by pcottle** [comment]
Yeah do you mind breaking this up into multiple PR's? Maybe on per file or class. We have a ton of incoming PR's so I think that'll be easier on everyone, since this will continue to get rebased on top of and go stale

all about more succinct python though :heart_eyes:

**Comment by pcottle** [comment]
(Closing for now, lets move to multiple PR's)
